src/main/java/org/prebid/server/floors/PriceFloorFetcher.java

```diff
@@ -0,0 +1,320 @@
+import org.prebid.server.settings.model.Account;
+import org.prebid.server.settings.model.AccountAuctionConfig;
+import org.prebid.server.settings.model.AccountPriceFloorsConfig;
+import org.prebid.server.settings.model.AccountPriceFloorsFetchConfig;
+import org.prebid.server.util.HttpUtil;
+import org.prebid.server.util.ObjectUtil;
+import org.prebid.server.vertx.http.HttpClient;
+import org.prebid.server.vertx.http.model.HttpClientResponse;
+
+import java.math.BigDecimal;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class PriceFloorFetcher {
+
+    private static final Logger logger = LoggerFactory.getLogger(PriceFloorFetcher.class);
+
+    private static final int ACCOUNT_FETCH_TIMEOUT_MS = 5000;
+    private static final int MAXIMUM_CACHE_SIZE = 300;
+
+    private final ApplicationSettings applicationSettings;
+    private final Metrics metrics;
+    private final Vertx vertx;
+    private final TimeoutFactory timeoutFactory;
+    private final HttpClient httpClient;
+    private final JacksonMapper mapper;
+
+    private final Set<String> fetchInProgress;
+    private final Map<String, AccountFetchContext> fetchedData;
+
+    public PriceFloorFetcher(ApplicationSettings applicationSettings,
+                             Metrics metrics,
+                             Vertx vertx,
+                             TimeoutFactory timeoutFactory,
+                             HttpClient httpClient,
+                             JacksonMapper mapper) {
+        this.applicationSettings = Objects.requireNonNull(applicationSettings);
+        this.metrics = Objects.requireNonNull(metrics);
+        this.vertx = Objects.requireNonNull(vertx);
+        this.timeoutFactory = Objects.requireNonNull(timeoutFactory);
+        this.httpClient = Objects.requireNonNull(httpClient);
+        this.mapper = Objects.requireNonNull(mapper);
+
+        fetchInProgress = new ConcurrentHashSet<>();
+        fetchedData = Caffeine.newBuilder()
+                .maximumSize(MAXIMUM_CACHE_SIZE)
+                .<String, AccountFetchContext>build()
+                .asMap();
+    }
+
+    public PriceFloorRules fetch(Account account) {
+        final AccountFetchContext accountFetchContext = fetchedData.get(account.getId());
+
+        return accountFetchContext != null
+                ? accountFetchContext.getRules()
+                : fetchPriceFloorRules(account);
+    }
+
+    private PriceFloorRules fetchPriceFloorRules(Account account) {
+        if (account == null) {
+            return null;
+        }
+
+        final AccountPriceFloorsFetchConfig fetchConfig = getFetchConfig(account);
+        final Boolean fetchEnabled = ObjectUtil.getIfNotNull(fetchConfig, AccountPriceFloorsFetchConfig::getEnabled);
+        final String fetchUrl = ObjectUtil.getIfNotNull(fetchConfig, AccountPriceFloorsFetchConfig::getUrl);
```

Comment:
Small refactoring. Seems `fetchEnabled` and `fetchUrl` variables can be encapsulated in `shouldFetchRules` mtehod.